### DISCUSSION TITLE
Nuclei JSON output

### DISCUSSION THREAD

**Discussion by w-dan** [projectdiscovery/nuclei#2005] [Ideas]
Hello,

I've been using Nuclei to automate some pentesting processes and I'm loving it.
To be precise, I've been using the JSON output feature and doing some queries on the generated documents, but I've come across a major issue (could be a functionality/feature request but I wasn't 100% sure so I started a discussion instead).

My problem is I'm trying to filter out certain type of detected vulnerabilities from the JSON file and extract some information about them and print them into an HTML file to make it readable, but there are some fields which contain arrays. As it happens, these arrays aren't generated in a "key":"value" format, but just "value" format instead, which makes JSON queries almost impossible in many cases with most JSON to HTML libraries.

Current output example:

```
{
    "template-id": "CVE-2016-6210",
    "info": {
      "name": "OpenSSH username enumeration < v7.3",
      "tags": [
        "cve",
        "cve2016",
        "network",
        "openssh"
      ],

   ...etc
```

For automation and JSON query purposes, it would be great if the output had a "key":"value" format instead, such as:

```
{
    "template-id": "CVE-2016-6210",
    "info": {
      "name": "OpenSSH username enumeration < v7.3",
      "tags": [
        "tag":"cve",
        "tag":"cve2016",
        "tag":"network",
        "tag":"openssh"
      ],
   ...
```
For every array, affected fields are: tags, reference, classification.cve-id, and classification.cwe-id

Thanks in advance

**Comment by ehsandeep** [comment]
@w-dan, thanks for opening the discussion; this sounds more like an issue with the library you are using. It should be simple enough to handle arrays in json; for example, try using https://jsongrid.com/json-grid

<img width="1915" alt="image" src="https://user-images.githubusercontent.com/8293321/168792738-c1595bea-dd18-4969-987d-6b557946e048.png">

Defiantly it would be easier to deal with `key:value` formatted data, but that also adds lots of redundant data in the json output, which can be avoided.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
